### PR TITLE
Write `adb logcat` results and screenshots to `FLUTTER_LOGS_DIR` if able.

### PR DESCRIPTION
**Pull request by matanlurey**
Fixes https://github.com/flutter/flutter/issues/143586.

This should make it easier to understand what is/isn't happening, and will unblock https://github.com/flutter/flutter/issues/143458, and make https://github.com/flutter/flutter/issues/143459 potentially easier to debug.

### CODE REVIEW

File: testing/scenario_app/android/README.md [hunk 1]
```diff
@@ -16,23 +16,46 @@ $ ./testing/scenario_app/run_android_tests.sh android_debug_unopt
...
+`--smoke-test` argument to run a single test by class name, which can be useful
+to verify the setup:
+
+```sh
+# From the root of the engine repository
+$ ./testing/scenario_app/run_android_tests.sh android_debug_unopt_arm64 --smoke-test dev.flutter.scenarios.EngineLaunchE2ETest
+```
+
+The result of `adb logcat` and screenshots taken during the test will be stored
+in a logs directory, which is either `FLUTTER_LOGS_DIR` (on CI) or locally in
```
Comment (reidbaker) [inline testing/scenario_app/android/README.md:36]: nonblocking nit: instead of (on ci) I think it should be (if set like in ci)

**Review by reidbaker — APPROVED** [review]

File: testing/scenario_app/android/README.md [hunk 2]
```diff
@@ -16,23 +16,46 @@ $ ./testing/scenario_app/run_android_tests.sh android_debug_unopt
...
 | 28          | Impeller (OpenGLES) | [Android 28 + Impeller OpenGLES][skia-gold-impeller-opengles-28] | Older Android devices (without `ImageReader`) on Impeller. |
-| 34          | Skia                | [Android 34 + Skia][skia-gold-skia-34]                           | Newer Android devices on Skia.                             |
-| 34          | Impeller (OpenGLES) | [Android 34 + Impeller OpenGLES][skia-gold-impeller-opengles-34] | Newer Android devices on Impeller with OpenGLES.           |
-| 34          | Impeller (Vulkan)   | [Android 34 + Impeller Vulkan][skia-gold-impeller-vulkan-34]     | Newer Android devices on Impeller.                         |
-
 [skia-gold-skia-28]: https://flutter-engine-gold.skia.org/search?left_filter=AndroidAPILevel%3D28%26GraphicsBackend%3Dskia&negative=true&positive=true&right_filter=AndroidAPILevel%3D28%26GraphicsBackend%3Dskia
 [skia-gold-impeller-opengles-28]: https://flutter-engine-gold.skia.org/search?left_filter=AndroidAPILevel%3D28%26GraphicsBackend%3Dimpeller-opengles&negative=true&positive=true&right_filter=AndroidAPILevel%3D28%26GraphicsBackend%3Dimpeller-opengles
+-->
+
```
Comment (reidbaker) [inline testing/scenario_app/android/README.md:52]: Did you mean to move this? if so ignore this comment.

**Review by reidbaker — APPROVED** [review]

File: testing/scenario_app/android/README.md (hunk 2 above)
Comment (matanlurey) [reply testing/scenario_app/android/README.md:52]: Yeah unfortunately markdown tables break with inline comments. Was intentional.

File: testing/scenario_app/android/README.md (hunk 1 above)
Comment (matanlurey) [reply testing/scenario_app/android/README.md:36]: Done

**Comment by matanlurey** [comment]
Rebased. @gaaclarke PTAL!

**Review by gaaclarke — APPROVED** [review]
LGTM!

**Pull request merged**